### PR TITLE
GH-768 AmCharts Simple Selection

### PR DESCRIPTION
**Pull request by mlzummo**
@GordonSmith @dtsnell4 @jbrundage need suggestions please

Adds AmCharts simple selection to Bar, Funnel, Pie

Fixes GH-768

Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

**Comment by dtsnell4** [comment]
Would it be a good idea to make the Pie widget a little smaller so the pieces don't get clipped buy the edge of the container when selected?

**Comment by mlzummo** [comment]
to keep track of what bar column has been colored going to add a .attr("data-selected","true");

**Comment by mlzummo** [comment]
yes im either going to do that or do a color change (if possible) ... since as of right now i cant emulate hpcc method of doing selection

**Comment by jbrundage** [comment]
I think Gordon's intention was that we add **Bag.SimpleSelection** to these chart widgets.

**Comment by mlzummo** [comment]
this._chart.addListener("clickGraphItem", function(e) {
    //var columnBorderSelection = d3.select(e.item.columnGraphics.node).select("path");
    //var borderWidth = parseInt(columnBorderSelection.attr("stroke-width"));
    //columnBorderSelection.attr("stroke-width",borderWidth+2);

    //e.item.dataContext[e.graph.lineColorField] = "#0079DC";

    var field = e.graph.fillColorsField;
    var data = e.item.dataContext;

    if (data[field]) {
        delete data[field];
    } else {
        data[field] = "#0079DC";
    }

    e.chart.validateData();

    context.click(context.rowToObj(context.data()[e.index]), context.columns()[e.target.columnIndex+1]);
});

File: src/amchart/CommonSerial.js [hunk 1]
```diff
@@ -512,6 +512,14 @@
         }
         this._chart = AmCharts.makeChart(domNode, initObj);
         this._chart.addListener("clickGraphItem", function(e) {
+            //var columnBorderSelection = d3.select(e.item.columnGraphics.node).select("path");
+            //var borderWidth = parseInt(columnBorderSelection.attr("stroke-width"));
+            //columnBorderSelection.attr("stroke-width",borderWidth+2);
+            
+            //e.item.dataContext[e.graph.lineColorField] = "#0079DC";
```
Comment (GordonSmith) [inline src/amchart/CommonSerial.js]: remove commented code.

File: src/amchart/Pie.js [hunk 2]
```diff
@@ -73,11 +80,13 @@
...
+        this._chart.colorField = "sliceColor";
+
         this._chart.dataProvider = this.formatData(this.data());
 
-        this._chart.colors = this.data().map(function (row) {
-            return this._palette(row[0]);
-        }, this);
+        // this._chart.colors = this.data().map(function (row) {
+        //     return this._palette(row[0]);
+        // }, this);
```
Comment (GordonSmith) [inline src/amchart/Pie.js]: remove commented code.

File: src/amchart/Pie.js [hunk 3]
```diff
@@ -123,6 +133,10 @@
         }
         this._chart = AmCharts.makeChart(domNode, initObj);
         this._chart.addListener("clickSlice", function(e) {
+            console.log(e);
```
Comment (GordonSmith) [inline src/amchart/Pie.js]: remove console.log

**Pull request closed without merging**